content/posts/batching-requests.md

```diff
@@ -0,0 +1,186 @@
+draft: true
+ShowToc: true
+TocOpen: true
+ShowRelatedContent: false
+summary: "A tutorial on using APISIX's request batching capability in Kubernetes Ingress."
+tags: ["ingress", "kubernetes", "apache apisix", "cloud-native"]
+categories: ["API Gateway"]
+series: ["Hands-On With Apache APISIX Ingress"]
+cover:
+  image: "/images/batching-requests/candy-banner.jpg"
+  alt: "Photo of candy grouped by colors."
+  caption: "Batching requests can reduce the number of client-server request/response cycles."
+  relative: false
+---
+
+With APISIX's request batching capability, you can group multiple API requests from the client into a single request. APISIX then turns this request into multiple requests to your upstream services, aggregates the response, and sends it back to the client as a single response.
+
+Batching can be useful in many practical scenarios where multiple requests are needed to fulfill a task. For example, the [Conference API](https://conferenceapi.azurewebsites.net/) exposes two different endpoints, `/speaker/{speakerId}/sessions` and `/speaker/{speakerId}/topics`, for listing the sessions and topics belonging to a speaker, respectively. You can make two API calls to show both in a front-end application.
+
+{{< figure src="/images/batching-requests/without-apisix.png#center" title="Making two requests" caption="Sometimes this might not be ideal." link="/images/batching-requests/without-apisix.png" target="_blank" class="align-center" >}}
+
+This might not always be ideal. Instead, you can use APISIX's request batching capability to group these requests into one, reducing the number of client-server calls.
+
+{{< figure src="/images/batching-requests/with-apisix.png#center" title="Batching requests" caption="Batching requests reduces the number of client-server calls." link="/images/batching-requests/with-apisix.png" target="_blank" class="align-center" >}}
+
+In this article, we will look at configuring this through APISIX Ingress in Kubernetes. If you are using the Admin API instead, you can check [this article](https://api7.ai/blog/batch-request-processing-with-api-gateway).
+
+## Set Up the Example
+
+First, let's set up the Conference API as our upstream. Like we did in the [previous article](/posts/apisix-ingress-external/), we will configure this external service in our upstream ([ApisixUpstream](https://apisix.apache.org/docs/ingress-controller/concepts/apisix_upstream/)) as shown below:
+
+```yaml {title="conference-api-upstream.yaml"}
+apiVersion: apisix.apache.org/v2
+kind: ApisixUpstream
+metadata:
+  name: conference-api-upstream
+spec:
+  externalNodes:
+    - type: Domain
+      name: conferenceapi.azurewebsites.net
+```
+
+## Enable Batching
+
+Request batching in APISIX is achieved through the [batch-requests](https://apisix.apache.org/docs/apisix/plugins/batch-requests/) plugin. You can enable the plugin by adding it to your Helm values file, as shown below:
+
+```yaml {title="values.yaml"}
+plugins:
```

Comment:
Does it add to the plugin or does it replace the entire plugins list like in the standalone?

If the latter, please tell it in bold, as it can be a source of surprise